### PR TITLE
Mileston: Todo app :memo: 

### PR DESCRIPTION
**Pull request by ahmedeid6842**
# Building a Todo List App with React 📝🚀

In this project, I created a Todo List app using React components to manage tasks. The app features some basic styling to improve the user experience. 🎨 I had a lot of fun building it and I'm excited to share it with you. 💻 I hope you find it useful and enjoyable to use. 😊 Let's get productive! 🙌🏼 

## Done:
- [x] Implemented a main component called `TodoContainer` to manage the todo list. 📌
- [x] Created a `TodoInput` component to enable users to add new tasks to the list. 📝
- [x] Developed a `TodoList` component and a `TodoItem` component to display the tasks in the list. 📋💡

### CODE REVIEW

File: src/components/TodoContainer.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  const getLocalTodos = () => {
+    if (!localStorage.getItem("todos")) {
+      localStorage.setItem("todos", JSON.stringify([]));
+    } else {
+      const localTodos = JSON.parse(localStorage.getItem("todos"));
+      setTodos(localTodos);
+    }
+  };
```
Comment (sergio-usma) [inline src/components/TodoContainer.jsx:28]: Hi! @ahmedeid6842 good job! I like the UI and the integration with the localStorage to save the To-Do's, the perform of your code is very clear and clean.

File: src/components/TodoInput.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    
+  function handleInputChange(event) {
+    setInputText(event.target.value);
+  }
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (!inputText) {
+      return;
+    }
```
Comment (sergio-usma) [inline src/components/TodoInput.jsx:16]: The use of handle events are identified and according to the project requirements!

File: src/components/TodoInput.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        placeholder="Please add your task"
+        value={inputText}
+        onChange={handleInputChange}
+        className="text-input"
+      />
+    </form>
+  );
```
Comment (sergio-usma) [inline src/components/TodoInput.jsx:38]: Well done using the form tag to add/create new notes in the To-Do list.

File: src/components/TodoItem.jsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+const TodoItem = ({ todo, todos, setTodos }) => {
+  const [isRemoved, setIsRemoved] = useState(false);
+
+  const deleteHandler = () => {
+    setIsRemoved(true);
+    setTimeout(() => {
+      setTodos(todos.filter((task) => task.id !== todo.id));
+    }, 800);
+  };
```
Comment (sergio-usma) [inline src/components/TodoItem.jsx:13]: Good implementation of the filter() method to delete To-Do's elements.

File: package.json [hunk 5]
```diff
@@ -2,20 +2,24 @@
...
     "web-vitals": "^2.1.4"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "predeploy": "npm run build",
+    "deploy": "gh-pages -d build"
```
Comment (sergio-usma) [inline package.json:22]: Very well done using the gh-pages dependency

File: src/components/TodoContainer.jsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const TodoContainer = () => {
+  const [todos, setTodos] = useState([]);
+
+  useEffect(() => {
+    getLocalTodos();
+  }, []);
+
+  useEffect(() => {
+    saveLocalTodos();
+  });
```
Comment (sergio-usma) [inline src/components/TodoContainer.jsx:15]: Well done using the useEffect() and hooks!

File: src/styles/TodoList.css [hunk 7]
```diff
@@ -0,0 +1,8 @@
+.todo-list {
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    list-style: none;
+    margin-top: 2rem;
+  }
```
Comment (sergio-usma) [inline src/styles/TodoList.css:7]: You could put all the styles in a single file for a small project like this!

**Review by sergio-usma — APPROVED** [review]
# STATUS: APPROVED ✅ 🍾
Hi @ahmedeid6842 
Your project is complete! There is nothing else to say other than... it's time to merge it!
Congratulations! 🎇

# To highlight 🍾
All linters are passing ✅
Good commit history ✅
Good PR description ✅ 
Good Testing implementation ✅ 

# Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application.

Cheers and Happy coding!

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmedeid6842** [comment]
I would like to express my gratitude to @sergio-usma  🙏 for taking the time ⏰ and making the effort 💪 to review my code. Additionally, I appreciate the valuable feedback provided. 🤗

**Pull request merged**